File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int answer;
    int targetNum;
    int size;
    public int solution(int[] numbers, int target) {
        answer = 0;
        targetNum = target;
        size = numbers.length;
        dfs(0, 0, numbers);
        return answer;
    }
    public void dfs(int cnt, int sum, int[] arr){
        
        if (cnt == size){
            if(sum == targetNum)
                answer++;
            return;
        }
        
        dfs(cnt + 1, sum + arr[cnt] , arr);
        dfs(cnt + 1, sum - arr[cnt] , arr);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = dfs(0, 0,numbers, target)
    return answer


def dfs(cur_num, index, numbers, target):
    if index == len(numbers):
        if cur_num == target:
            return 1
        return 0
    
    return dfs(cur_num+numbers[index], index+1, numbers, target) + dfs(cur_num-numbers[index], index+1, numbers, target)
    
    